extensions/spotify-player/src/player.tsx

```diff
@@ -0,0 +1,370 @@
+import {
+  Action,
+  ActionPanel,
+  Clipboard,
+  closeMainWindow,
+  Color,
+  environment,
+  getPreferenceValues,
+  Icon,
+  LaunchType,
+  List,
+  open,
+  showHUD,
+  showToast,
+  Toast,
+} from "@raycast/api";
+import { useEffect, useState } from "react";
+import BrowseAll from "./browseAll";
+import { getState, getTrack, nextTrack, pause, play, playPause, previousTrack } from "./spotify/applescript";
+import { likeCurrentlyPlayingTrack, startPlaySimilar, useSearch } from "./spotify/client";
+import { SpotifyPlayingState, SpotifyState, TrackInfo } from "./spotify/types";
+import NotInstalledDetail from "./components/NotInstalledDetail";
+import NowPlayingDetail from "./components/NowPlayingDetail";
+import NowPlayingEmptyDetail from "./components/NowPlayingEmptyDetail";
+import { SearchListItems } from "./components/SearchListItems";
+import FeaturedPlaylists from "./featuredPlaylists";
+import SearchAlbums from "./searchAlbums";
+import SearchArtists from "./searchArtists";
+import SearchPlaylists from "./searchPlaylists";
+import SearchTracks from "./searchTracks";
+import { isSpotifyInstalled, Preferences, showTrackNotification } from "./utils";
+
+export default function Player() {
+  const preferences = getPreferenceValues<Preferences>();
+
+  const [spotifyInstalled, setSpotifyInstalled] = useState<boolean | undefined>();
+  const [currentlyPlayingTrack, setCurrentlyPlayingTrack] = useState<TrackInfo | undefined>();
+  const [currentSpotifyState, setCurrentSpotifyState] = useState<SpotifyState | undefined>();
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchText, setSearchText] = useState<string>();
+
+  const response = useSearch(searchText);
+
+  const fetchPlayerAndTrackState = async () => {
+    // Check if Spotify is installed (only try this once)
+    let isInstalled = spotifyInstalled;
+    if (isInstalled == undefined) {
+      isInstalled = await isSpotifyInstalled();
+      setSpotifyInstalled(await isSpotifyInstalled());
+    }
+
+    // If Spotify is installed then fetch the player and track state
+    if (isInstalled) {
+      try {
+        const [state, track] = await Promise.all([getState(), getTrack()]);
+
+        setCurrentSpotifyState(state);
+        setCurrentlyPlayingTrack(track);
+      } catch (err) {
+        if (environment.launchType != LaunchType.Background) {
+          showToast(Toast.Style.Failure, String(err));
+        }
+      }
+    }
+
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    // Fetch initial data
+    fetchPlayerAndTrackState();
+
+    // Attempt to refresh the state every 5s
```

Comment:
In testing this seemed a reasonable interval but perhaps we should be more conservative?